src/trainer.py

```diff
@@ -1276,28 +1251,26 @@ def _restore_checkpoint(self, search_phases_in_priority_order=["pretrain"]):
         computation graph, you should use the native Pytorch functions:
         `` model.load_state_dict(torch.load("/path/to/model/weights.th"))``
 
+        We restore based on the phase. If phase=target_train, we look backwards based 
+        on the tasks to find the most recen.
         Returns
         -------
         epoch
             The epoch at which to resume training.
         """
-
-        suffix_to_load = self._find_last_checkpoint_suffix(
-            search_phases_in_priority_order=search_phases_in_priority_order
-        )
+        suffix_to_load = None
+        tasks = tasks[::-1]  # reverse the tasks
+        for task in tasks:
```

Comment:
This feels redundant with check_for_previous_checkpoints. Do we really need both?

Reply:
There's enough difference between the two, and check_for_previous_checkpoints callsfind_last_checkpoint_suffix. check_for_previous_checkpoints is used specifically in target_train - otherwise, we would have to put all the code in that function into main.py, which I was trying to avoid.

Reply:
I guess most of the difference strikes me as unnecessary, unless I'm missing something. There's some logic to prevent us from launching the trainer at all in target_train for finished tasks, but we might as well set that up for both phases if we're going to add it. 

If we check should_stop on each checkpoint to decide whether to launch the trainer, rather than just checking if files exist, we'll get a solution that works for both phases. We'd just need to open the tiny training_state file, not the huge model/optimizer state.

Reply:
(I know it'd be quicker to just push this as soon as it's working and reasonably well commented, but debugging has gotten so tricky that I'd like to be _really_ careful about increasing code complexity, even if it means a small delay.)